inductiva/_benchmarks/benchmark.py

```diff
@@ -144,83 +145,87 @@ def export(
-        """
-        data = {}
-        tasks = self.get_tasks()
-        for task in tasks:
-            info = task.get_info()
-            vm_type = info.executer.vm_type
-            data.setdefault(vm_type, [])
-            data[vm_type].append({
-                "task_id": info.task_id,
-                "estimated_computation_cost": info.estimated_computation_cost,
-                "computation_time": info.time_metrics.computation_seconds.value,
-            })
-        return data
+        Gathers the configuration and performance metrics for each run
+        associated with the benchmark in a list, including computation cost and
+        execution time.
 
-    def gather_detailed_data(self) -> dict:
-        """
-        Gathers comprehensive information about all tasks associated with the 
-        benchmark, including additional performance metrics and task metadata.
+        This method retrieves input parameters and task information for each
+        run, and optionally summarizes the results by filtering only the
+        parameters that vary between runs.
+        
+        Args:
+            summary (bool): If True, only the parameters that vary between 
+                different runs are included in the result, providing a
+                summary of the information for each run. Defaults to True.
 
         Returns:
-            dict: A dictionary organized by virtual machine type, containing
-                detailed information about each task associated with the
-                benchmark.
+            list: A list containing the configuration and performance 
+                metrics for each run.
         """
-        data = {}
-        tasks = self.get_tasks()
-        for task in tasks:
-            info = task.get_info()
-            vm_type = info.executer.vm_type
-            data.setdefault(vm_type, [])
+
+        def get_task_input_params(task):
             input_filename = "input.json"
             input_dir_path = task.download_inputs(filenames=[input_filename])
             input_file_path = input_dir_path.joinpath(input_filename)
             with open(input_file_path, mode="r", encoding="utf-8") as file:
-                input_json = json.load(file)
-            data[vm_type].append({
-                "task_info": info.to_dict(),
-                "task_input_metadata": input_json,
+                return json.load(file)
+
+        def summarize(info):
+            attrs_lsts = defaultdict(list)
+            for attrs in info:
+                for k, v in attrs.items():
+                    attrs_lsts[k].append(v)
+            filtered = {k for k, v in attrs_lsts.items() if len(set(v)) > 1}
+            return [{key: attrs[key] for key in filtered} for attrs in info]
+
+        info = []
+        tasks = self.get_tasks()
+        for task in tasks:
+            task_input_params = get_task_input_params(task)
+            task_info = task.get_info()
```

Comment:
You can access `task.info` that is filled by `get_tasks` instead. Otherwise, `get_info` will result in a new API request to fetch the same information that was already present.

Reply:
I'm not sure if it's safe to assume that the cached task information is the most recent version. For example, there's no guarantee that the task has been completed yet. 

Users that call ```export``` multiple times should see updated values, not cached values.

Reply:
The information is not cached, it is fetched immediately before in line 216.